src/opsinputs/opsinputs_fill_mod.F90

```diff
@@ -1236,28 +1258,25 @@ subroutine opsinputs_fill_fillreal2d( &
 integer(integer64), intent(in)                      :: VarobsLength
 character(len=*), intent(in)                        :: JediVarName
 character(len=*), intent(in)                        :: JediVarGroup
-type(opsinputs_channeloffset), optional, intent(in) :: OffsetChans
-logical, optional, intent(in)                       :: useActualChans
+logical, optional, intent(in)                       :: compressVarChannels
+integer(integer64), optional, intent(in)            :: sizeVarObs
+integer(c_int), optional, intent(in)                :: varChannels(:)
+
+
 
 ! Body:
 if (JediToOpsLayoutMapping % ConvertRecordsToMultilevelObs) then
   call opsinputs_fill_fillreal2d_records( &
     Hdr, OpsVarName, JediToOpsLayoutMapping, Real2, ObsSpace, VarobsLength, JediVarName, JediVarGroup)
 else
-  if (Present(OffsetChans)) then
-    if (Present(useActualChans)) then
-      call opsinputs_fill_fillreal2d_norecords( &
-        Hdr, OpsVarName, JediToOpsLayoutMapping % NumOpsObs, Real2, ObsSpace, Channels, &
-        JediVarName, JediVarGroup, OffsetChans, useActualChans)
-    else
-      call opsinputs_fill_fillreal2d_norecords( &
-        Hdr, OpsVarName, JediToOpsLayoutMapping % NumOpsObs, Real2, ObsSpace, Channels, &
-        JediVarName, JediVarGroup, OffsetChans)
-    end if
+  if (size(varChannels) > 0) then
+    call opsinputs_fill_fillreal2d_norecords( &
+      Hdr, OpsVarName, JediToOpsLayoutMapping % NumOpsObs, Real2, ObsSpace, Channels, &
+      JediVarName, JediVarGroup, compressVarChannels, sizeVarObs, varChannels)
```

Comment:
As `varChannels` , `sizeVarObs` and `varChannels` are all optional this code is currently not safe.  With optional there needs to be a present check else you can't guarantee memory integrity.

Reply:
fixed